### PR TITLE
feat: #2445 graphql server for documents

### PR DESCRIPTION
**Pull request by ant066**
# Pull request checklist

## Does this close any currently open issues?

fixes: #2445

**Please check if your PR fulfills the following requirements:**

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint`) has passed locally and any fixes were made for failures
- [x] Test (`yarn test`) has passed locally and any fixes were made for failures

## Pull request type

**Please check the type of change your PR introduces:**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #2445

## What is the new behavior?

-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

According to Graphql spec, the response of graphql service should be primitives values. As I saw `Download document endpoint` return a blob file (maybe too large), it not suitable be the response of graphql service. So that why I am not expanding it in the graphql server.

### CODE REVIEW

File: packages/graphql-server/src/resolvers/documents/types.graphql [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+type FileData {
+  base64: String
+}
+
+type Query {
+  GetDocuments(
+    pageSize: Int
+    pageNumber: Int
+    sortBy: String
+    embed: [String]
```
Comment (vuhuucuong) [inline packages/graphql-server/src/resolvers/documents/types.graphql:51]: This should be an `enum` type like this
https://github.com/reapit/foundations/pull/2455/files#diff-e6bfd44f89f7928415ab41f135fe56f4

**Comment by vuhuucuong** [comment]
Can you scan through all the `types.graphql` file and change all the `embed document` from `JSON` to `DocumentModel`
For example, this file https://github.com/reapit/foundations/pull/2455/files#diff-e6bfd44f89f7928415ab41f135fe56f4R28-R33
should be changed from `documents: JSON` to `documents: [DocumentModel]`.
- Everything with `s` at the end should be wrapped in `[]`
- if it without `s` it should be `document: DocumentModel`

File: packages/graphql-server/src/resolvers/documents/types.graphql [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    id: [String]
+    associatedId: [String]
+    associatedType: [String]
+    typeId: [String]
+    metadata: [String]
+  ): PagedResultDocumentModel!
+  GetDocumentById(id: String!, embed: [String]): DocumentModel!
```
Comment (vuhuucuong) [inline packages/graphql-server/src/resolvers/documents/types.graphql:58]: This `embed` should be `enum` too

**Comment by ant066** [comment]
> Can you scan through all the `types.graphql` files and change all the `embed document(s)` from `JSON` to `DocumentModel`
> For example, this file https://github.com/reapit/foundations/blob/master/packages/graphql-server/src/resolvers/contacts/types.graphql#L29
> should be changed from `documents: JSON` to `documents: [DocumentModel]`.
> 
> * Everything with `s` at the end should be wrapped in `[]`
> * if it doesn't have `s`, it should be `document: DocumentModel`

will do that. Thanks @vuhuucuong

**Review by nphivu414 — APPROVED** [review]

**Review by phmngocnghia — APPROVED** [review]

**Pull request merged**